contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -125,7 +136,8 @@ contract OETHVaultCore is VaultCore {
             "Redeem amount lower than minimum"
         );
 
-        if (IERC20(weth).balanceOf(address(this)) >= amountMinusFee) {
+        // If there is any WETH in the Vault available after accounting for the withdrawal queue
+        if (_wethAvailable() >= amountMinusFee) {
```

Comment:
🟥 This appears to bypass our check of only giving out available (unreserved) funds.

If we don't have funds available, the redeem should fail, not collect funds to give the user from another source.

Reply:
I think we simplify redeem and don't try and pull funds from the default strategy. If there's not enough available funds in the vault after accounting for the withdrawal queue, we will revert with `Liquidity error`.
If the user really wants to, they can request a withdrawal.

Reply:
I'm thinking our casts in general, not just this one line. Basically unless we have a serious reason for not doing a safe cast, we should use a safe cast. Gas overhead is minimal, though I'm sad it makes the code uglier. Maybe a `using` would help?

Reply:
The changes have been pushed

Reply:
For the review of this bug, here's the relevant section in `_redeem`

```Solidity 
// If there is any WETH in the Vault available after accounting for the withdrawal queue
if (_wethAvailable() >= amountMinusFee) {
    // Use Vault funds first if sufficient
    IERC20(weth).safeTransfer(msg.sender, amountMinusFee);
} else {
    address strategyAddr = assetDefaultStrategies[weth];
    if (strategyAddr != address(0)) {
        // Nothing in Vault, but something in Strategy, send from there
        IStrategy strategy = IStrategy(strategyAddr);
        strategy.withdraw(msg.sender, weth, amountMinusFee);
    } else {
        // Cant find funds anywhere
        revert("Liquidity error");
    }
}
```

Here's the new implementation of this section
```Solidity
// Is there enough WETH in the Vault available after accounting for the withdrawal queue
require(_wethAvailable() >= amountMinusFee, "Liquidity error");

// Transfer WETH minus the fee to the redeemer
IERC20(weth).safeTransfer(msg.sender, amountMinusFee);
```